File: m-solutions2020/m-solutions2020_c.cs
// https://atcoder.jp/contests/m-solutions2020/submissions/15466564

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var input = Console.ReadLine().Split().Select(int.Parse).ToArray();
        var K = input[1];
        var A = Console.ReadLine().Split().Select(int.Parse).ToArray();

        for (int i = K; i < A.Length; i++)
        {
            if (A[i - K] < A[i])
            {
                Console.WriteLine("Yes");
            }
            else
            {
                Console.WriteLine("No");
            }
        }
    }
}

File: m-solutions2020/m-solutions2020_c.fs
// https://atcoder.jp/contests/m-solutions2020/submissions/15465089

let [| N; K |] = stdin.ReadLine().Split() |> Array.map int
let A = stdin.ReadLine().Split() |> Array.map int

[| K..(N - 1) |]
|> Array.map (fun x ->
    if A.[x - K] < A.[x] then "Yes"
    else "No")
|> Array.iter stdout.WriteLine
